### ISSUE TITLE
Better way to create blocks in CPP macros

### ISSUE THREAD

**Issue by JBlaschke** [Trzs/kokkos_ifelse#1]
I use "better" in the sense of improved predictability of code -- this has all the smell of a hack though. Sadly there is no standard to defined a multi-statement CPP macro, so we're all stuck with this "official" way of doing things.

In your `LOOK_INTO` macro,
https://github.com/Trzs/kokkos_ifelse/blob/e5a4248bc8863ae8df0b9ef45d220f64771cd5db/main.cpp#L4
consider doing this:
```c
#define LOOK_INTO(var) do { printf("%d (%s): %.9E \n", __LINE__, #var, var); } while(0)
```
or dropping the curly braces altogether -- I assume the curly braces are a hold-over from the time you where checking pixel index. If you end up writing multi-statement macros, you should fence them with a `do { ... } while (0)`

TL;DR -- the do statement converts the `{...}` (a block) into a single statement. Most compilers are smart enough to realize the constant in the `while (0)` part, and will elide any jump/branch/goto.

**Comment by Trzs** [comment]
Fixed in 0cee38add5a255c5d3093780fa43589e796dd80a. Thanks!
